javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1194 @@
+              throw new IllegalStateException
+            _hasNext = false
+            o
+          }
+
+          def remove(): Unit =
+            throw new UnsupportedOperationException
+        }
+      }
+    })
+  }
+
+  def singletonList[T](o: T): List[T] = {
+    unmodifiableList(new AbstractList[T] with Serializable {
+      def size(): Int = 1
+
+      def get(index: Int): T =
+        if (index === 0) o
+        else throw new IndexOutOfBoundsException(index.toString)
+    })
+  }
+
+  def singletonMap[K, V](key: K, value: V): Map[K, V] = {
+    unmodifiableMap[K, V](new AbstractMap[K, V] with Serializable {
+      def entrySet(): Set[Map.Entry[K, V]] =
+        singleton(new AbstractMap.SimpleImmutableEntry(key, value))
+    })
+  }
+
+  def nCopies[T](n: Int, o: T): List[T] = {
+    if (n < 0)
+      throw new IllegalArgumentException
+
+    unmodifiableList(new AbstractList[T] with Serializable with RandomAccess {
+      def size(): Int = n
+
+      def get(index: Int): T = {
+        if (index < 0 || index >= n)
+          throw new IndexOutOfBoundsException
+        o
+      }
+    })
+  }
+
+  def reverseOrder[T](): Comparator[T] =
+    reverseOrder(naturalComparator[T])
+
+  def reverseOrder[T](cmp: Comparator[T]): Comparator[T] = {
+    class ReverseComparator extends Comparator[T] {
+      override def compare(o1: T, o2: T): Int = cmp.compare(o2, o1)
+    }
+    cmp match {
+      case _: Serializable => new ReverseComparator with Serializable
+      case _                  => new ReverseComparator
+    }
+  }
+
+  def enumeration[T](c: Collection[T]): Enumeration[T] = {
+    val it = c.iterator
+    new Enumeration[T] {
+      override def hasMoreElements: Boolean =
+        it.hasNext
+
+      override def nextElement(): T =
+        it.next()
+    }
+  }
+
+  def list[T](e: Enumeration[T]): ArrayList[T] = {
+    val arrayList = new ArrayList[T]
+    e.foreach(arrayList += _)
+    arrayList
+  }
+
+  def frequency(c: Collection[_], o: AnyRef): Int =
+    c.count(_ === o)
+
+  def disjoint(c1: Collection[_], c2: Collection[_]): Boolean = {
+    if (c1.size < c2.size)
+      !c1.exists(elem => c2.contains(elem))
+    else
+      !c2.exists(elem => c1.contains(elem))
+  }
+
+  def addAll[T](c: Collection[_ >: T], elements: Array[AnyRef]): Boolean =
+    c.addAll(asJavaCollection(elements.asInstanceOf[Array[T]]))
+
+  def newSetFromMap[E](map: Map[E, Boolean]): Set[E] = {
+    if (map.nonEmpty)
+      throw new IllegalArgumentException
+
+    new WrappedSet[E, Set[E]] {
+      override protected val inner: Set[E] =
+        map.keySet
+
+      override def add(e: E): Boolean =
+        map.put(e, true) === null
+
+      override def addAll(c: Collection[_ <: E]): Boolean =
+        c.foldLeft(false)((prev, elem) => map.put(elem, true) === null || prev)
+    }
+  }
+
+  @inline
+  private def modulo(a: Int, b: Int): Int = ((a % b) + b) % b
+
+  @inline
+  private def naturalComparator[T]: Comparator[T] = {
```

Comment:
Ah OK. Well then maybe just create a `new Comparator[T]`, and cast `o1.asInstanceOf[jl.Comparable[T]]` when doing the `.compareTo(o2)`. That way, the "real" cast (the one that is checked) is more apparent. (Since `.asInstanceOf[Comparator[T]]` is erased.